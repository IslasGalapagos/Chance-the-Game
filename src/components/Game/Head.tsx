/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright Â© 2018 Evgeny Sysoletin. All rights reserved.
*/

import * as React from 'react';
import {StyledWrapper, View} from './Head.styles';
import {Positions} from '.';

import '../../static/head-default.svg';
import '../../static/head-happy.svg';
import '../../static/head-unhappy.svg';
import '../../static/view.svg';

export const enum Emotions {
  Default = 'default',
  Happy = 'happy',
  Unhappy = 'unhappy'
}

export interface Props {
  position: Positions;
  isMouseEnter: boolean;
  isWin: boolean | null;
}

class Head extends React.PureComponent<Props, {}> {
  constructor(props: any) {
    super(props);

    this.getEmotion = this.getEmotion.bind(this);
  }

  getEmotion(): Emotions {
    switch (this.props.isWin) {
      case true:
        return Emotions.Happy;
        break;
      case false:
        return Emotions.Unhappy;
        break;
      case null:
      default:
        return Emotions.Default;
    }
  }

  render() {
    const {position, isMouseEnter} = this.props;
    const emotion: Emotions = this.getEmotion();

    return (
      <StyledWrapper position={position} emotion={emotion}>
        {isMouseEnter && (
          <View src="./images/view.svg" emotion={emotion} alt="view" />
        )}
        <img src={`./images/head-${emotion}.svg`} alt="head" />
      </StyledWrapper>
    );
  }
}

export default Head;
