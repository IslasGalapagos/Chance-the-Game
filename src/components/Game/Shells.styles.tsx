/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright Â© 2018 Evgeny Sysoletin. All rights reserved.
*/

import styled, {css, keyframes} from 'react-emotion';

export const StyledWrapper = styled('div')`
  position: relative;

  display: block;
`;

const shellHoverKeyframes = keyframes`
  from {
    transform: scale(1);
  }
  50% {
    transform: scale(0.95, 1.02);
  }
  to {
    transform: scale(1);
  }
`;

const ShellCSS = css`
  display: inline-block;
  vertical-align: bottom;

  cursor: pointer;

  &:hover {
    animation: ${shellHoverKeyframes} 0.2s ease infinite;
    animation-delay: 0.2s;
  }
`;

interface StyledShellProps {
  isHidden: boolean;
  middle?: boolean;
}

export const StyledShell = styled('img')<StyledShellProps>`
  margin: ${({middle}) => (middle ? '0 50px' : '')};

  visibility: ${({isHidden}) => (isHidden ? 'hidden' : '')};

  ${() => ShellCSS};
`;
