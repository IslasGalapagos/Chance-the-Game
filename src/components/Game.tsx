/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright Â© 2018 Evgeny Sysoletin. All rights reserved.
*/

import * as React from 'react';
import shuffle from 'lodash-es/shuffle';
import StyledWrapper from './Game.styles';
import Head from './Head';
import Shells from './Shells';
import {
  ThunkSetScore,
  ThunkSetCoefficients,
  ThunkSetTotalScore
} from '../store/thunk-actions';

export const enum Positions {
  Left = 'left',
  Center = 'center',
  Right = 'right'
}

export interface Props {
  setScore: (isWin: boolean, prize: number) => ThunkSetScore;
  setTotalScore: (isWin: boolean, scoreNum: number) => ThunkSetCoefficients;
  setCoefficients: (isWin: boolean) => ThunkSetTotalScore;
}

export interface State {
  readonly position: Positions;
  readonly isMouseEnter: boolean;
  readonly chosenPosition: Positions | null;
  readonly isWin: boolean | null;
}

class Game extends React.PureComponent<Props, State> {
  static readonly SCORE_NUM: number = 50;
  static readonly PRIZE: number = 500;

  readonly state: State;

  constructor(props: any) {
    super(props);

    this.state = {
      position: Positions.Center,
      isMouseEnter: false,
      chosenPosition: null,
      isWin: null
    };

    this.move = this.move.bind(this);
    this.onChoose = this.onChoose.bind(this);
    this.onMouseLeave = this.onMouseLeave.bind(this);
    this.updateStore = this.updateStore.bind(this);
  }

  move(position: Positions): void {
    this.setState({
      position,
      isMouseEnter: true
    });
  }

  random(): Positions[] {
    const position: Positions[] = shuffle([
      Positions.Left,
      Positions.Center,
      Positions.Right
    ]);
    const oneOrTwo = Math.floor(Math.random() * 4 + 1) === 3 ? 2 : 1;

    return Array.apply(null, {length: oneOrTwo}).map(
      () => position.splice(0, 1)[0]
    );
  }

  onChoose(position: Positions): void {
    const winningPositions: Positions[] = this.random();
    const isWin = winningPositions.includes(position);

    this.setState({
      chosenPosition: position,
      isWin
    });

    setTimeout(() => {
      this.setState({
        chosenPosition: null,
        isWin: null
      });
    }, 800);

    this.updateStore(isWin);
  }

  updateStore(isWin: boolean): void {
    const {setScore, setTotalScore, setCoefficients} = this.props;

    setScore(isWin, Game.PRIZE);
    setTotalScore(isWin, Game.SCORE_NUM);
    setCoefficients(isWin);
  }

  onMouseLeave() {
    this.setState({
      isMouseEnter: false
    });
  }

  render() {
    const {position, isMouseEnter, chosenPosition, isWin} = this.state;

    return (
      <StyledWrapper>
        <Head isWin={isWin} position={position} isMouseEnter={isMouseEnter} />
        <Shells
          move={this.move}
          onChoose={this.onChoose}
          onMouseLeave={this.onMouseLeave}
          chosenPosition={chosenPosition}
          isWin={isWin}
        />
      </StyledWrapper>
    );
  }
}

export default Game;
