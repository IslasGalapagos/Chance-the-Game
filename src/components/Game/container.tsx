/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright Â© 2018 Evgeny Sysoletin. All rights reserved.
*/

import {connect} from 'react-redux';
import {ThunkDispatch} from 'redux-thunk';
import Game from '.';
import {
  checkAndSetScore,
  checkAndSetCoefficients,
  computeAndSetTotalScore
} from '../../store/thunk-actions';
import {setScore, setCoefficients, setTotalScore} from '../../store/actions';
import {State} from '../../store/reducer';

type Actions = ReturnType<
  typeof setScore | typeof setCoefficients | typeof setTotalScore
>;

const mapDispatchToProps = (dispatch: ThunkDispatch<State, null, Actions>) => ({
  setScore: (isWin: boolean, prize: number) =>
    dispatch(checkAndSetScore(isWin, prize)),
  setCoefficients: (isWin: boolean) => dispatch(checkAndSetCoefficients(isWin)),
  setTotalScore: (isWin: boolean, scoreNum: number) =>
    dispatch(computeAndSetTotalScore(isWin, scoreNum))
});

export default connect(
  undefined,
  mapDispatchToProps
)(Game);
