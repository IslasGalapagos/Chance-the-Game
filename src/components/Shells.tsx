/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright © 2018 Evgeny Sysoletin. All rights reserved.
*/

import * as React from 'react';
import {StyledWrapper, StyledShell} from './Shells.styles';
import Ball from './Ball';
import {Positions} from './Game';

import '../static/shell-1.svg';
import '../static/shell-2.svg';
import '../static/shell-3.svg';

export interface Props {
  move: (position: Positions) => void;
  onChoose: (position: Positions) => void;
  onMouseLeave: () => void;
  chosenPosition: Positions | null;
  isWin: boolean | null;
}

class Shells extends React.PureComponent<Props, {}> {
  constructor(props: any) {
    super(props);

    this.onChoose = this.onChoose.bind(this);
  }

  onChoose(position: Positions): () => void {
    return () => {
      this.props.onChoose(position);
    };
  }

  onMouseEnter(position: Positions): () => void {
    return () => {
      this.props.move(position);
    };
  }

  render() {
    const {onMouseLeave, chosenPosition, isWin} = this.props;

    const shells = [Positions.Left, Positions.Center, Positions.Right].map(
      (position, index) => (
        <StyledShell
          src={`./images/shell-${index + 1}.svg`}
          alt={`shell № ${index + 1}`}
          key={position}
          onClick={this.onChoose(position)}
          onMouseEnter={this.onMouseEnter(position)}
          onMouseLeave={onMouseLeave}
          isHidden={chosenPosition !== null && chosenPosition === position}
          middle={index === 1}
        />
      )
    );

    return (
      <StyledWrapper>
        {shells}
        {isWin !== null &&
          chosenPosition !== null && (
            <Ball isWin={isWin} chosenPosition={chosenPosition} />
          )}
      </StyledWrapper>
    );
  }
}

export default Shells;
