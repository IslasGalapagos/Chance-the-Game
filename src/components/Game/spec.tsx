/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright Â© 2018 Evgeny Sysoletin. All rights reserved.
*/

import {shallow} from 'enzyme';
import * as testRenderer from 'react-test-renderer';
import * as React from 'react';
import isEqual from 'lodash-es/isEqual';
import Game, {Positions, Props} from '.';
import Head from './Head';
import Shells from './Shells';

const positionsArr: Positions[] = [
  Positions.Left,
  Positions.Center,
  Positions.Right
];

const props: Props = {
  setScore: jest.fn(),
  setTotalScore: jest.fn(),
  setCoefficients: jest.fn()
};

describe('<Game/>', () => {
  beforeEach(() => {
    (props.setScore as any).mockClear();
    (props.setTotalScore as any).mockClear();
    (props.setCoefficients as any).mockClear();
  });

  it('renders correctly', () => {
    const treeAndStyles = testRenderer.create(<Game {...props} />).toJSON();
    expect(treeAndStyles).toMatchSnapshot();
  });

  // STATE
  it(`sets 'center' to state.position by default`, () => {
    const component = shallow(<Game {...props} />);
    expect(component.state('position')).toEqual(Positions.Center);
  });

  it(`sets false to state.isMouseEnter by default`, () => {
    const component = shallow(<Game {...props} />);
    expect(component.state('isMouseEnter')).toEqual(false);
  });

  it(`sets null to state.chosenPosition by default`, () => {
    const component = shallow(<Game {...props} />);
    expect(component.state('chosenPosition')).toBeNull();
  });

  it(`sets null to state.isWin by default`, () => {
    const component = shallow(<Game {...props} />);
    expect(component.state('isWin')).toEqual(null);
  });

  // METHODS
  it('has move that changes state.position depends on arg & sets true to state.isMouseEnter', () => {
    const component = shallow(<Game {...props} />);
    const {move} = component.instance() as Game;
    move(Positions.Left);
    expect(component.state('position')).toEqual(Positions.Left);
    expect(component.state('isMouseEnter')).toEqual(true);
  });

  it('has onMouseLeave that sets false to state.isMouseEnter', () => {
    const component = shallow(<Game {...props} />);
    const {onMouseLeave} = component.instance() as Game;
    component.setState({isMouseEnter: true});
    onMouseLeave();
    expect(component.state('isMouseEnter')).toEqual(false);
  });

  it('has random that returns random Position(s)', () => {
    const component = shallow(<Game {...props} />);
    const {random} = component.instance() as Game;
    const randomsArr: Positions[][] = Array.apply(null, {length: 20}).map(() =>
      random()
    );

    randomsArr.forEach((x: Positions[]) => {
      expect(positionsArr).toEqual(expect.arrayContaining(x));
    });

    const singles = randomsArr.filter((x: Positions[]) => x.length === 1);
    expect(singles.length).not.toEqual(0);
    const tuples = randomsArr.filter((x: Positions[]) => x.length === 2);
    expect(tuples.length).not.toEqual(0);

    expect(singles.length + tuples.length).toEqual(20);

    const isSinglesDiff = singles.some(
      (x: Positions[], index) => index > 0 && x[0] !== singles[index - 1][0]
    );
    expect(isSinglesDiff).toBeTruthy();

    const isTupleasDiff = tuples.some((x: Positions[], index) => {
      x.sort();

      if (index === 0) {
        return false;
      }

      return !isEqual(x[0], tuples[index - 1]);
    });

    expect(isTupleasDiff).toBeTruthy();
  });

  it(`has onChoose that takes user's choice, determines win or fail by random & changes state for some time`, () => {
    jest.useFakeTimers();

    const spyRandom = jest.spyOn(Game.prototype, 'random');
    const component = shallow(<Game {...props} />);
    const {onChoose} = component.instance() as Game;

    expect(spyRandom).not.toBeCalled();

    onChoose(Positions.Left);
    expect(spyRandom).toBeCalledTimes(1);
    expect(component.state('chosenPosition')).toEqual(Positions.Left);
    expect(component.state('isWin')).toEqual(expect.any(Boolean));

    jest.runAllTimers();

    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 800);
    expect(component.state('chosenPosition')).toBeNull();
    expect(component.state('isWin')).toBeNull();

    jest.restoreAllMocks();
  });

  it('has updateStore that calls props.updateScore, props.updateTotalScore and props.updateCoefficients', () => {
    const component = shallow(<Game {...props} />);
    const {updateStore} = component.instance() as Game;
    updateStore(true);
    expect(props.setScore).toBeCalledTimes(1);
    expect(props.setScore).toBeCalledWith(true, expect.any(Number));
    expect(props.setTotalScore).toBeCalledTimes(1);
    expect(props.setTotalScore).toBeCalledWith(true, Game.SCORE_NUM);
    expect(props.setCoefficients).toBeCalledTimes(1);
    expect(props.setCoefficients).toBeCalledWith(true);
  });

  it('calls updateStore in onChoose', () => {
    const spyUpdateStore = jest.spyOn(Game.prototype, 'updateStore');
    const spyRandom = jest.spyOn(Game.prototype, 'random');
    const component = shallow(<Game {...props} />);
    const {onChoose} = component.instance() as Game;
    onChoose(Positions.Left);
    expect(spyUpdateStore).toBeCalledTimes(1);
    expect(spyUpdateStore).toBeCalledWith(
      spyRandom.mock.results[0].value.includes(Positions.Left)
    );
  });

  // HEAD
  it(`sets state.isWin to Head's prop isWin`, () => {
    const component = shallow(<Game {...props} />);
    expect(component.find(Head).prop('isWin')).toEqual(
      component.state('isWin')
    );
    component.setState({isWin: true});
    expect(component.find(Head).prop('isWin')).toEqual(true);
  });

  it(`sets state.position to Head's prop position`, () => {
    const component = shallow(<Game {...props} />);
    expect(component.find(Head).prop('position')).toEqual(
      component.state('position')
    );
    component.setState({position: Positions.Left});
    expect(component.find(Head).prop('position')).toEqual(Positions.Left);
  });

  it(`sets state.isMouseEnter to Head's prop isMouseEnter`, () => {
    const component = shallow(<Game {...props} />);
    expect(component.find(Head).prop('isMouseEnter')).toEqual(
      component.state('isMouseEnter')
    );
    component.setState({isMouseEnter: true});
    expect(component.find(Head).prop('isMouseEnter')).toEqual(true);
  });

  // SHELLS
  it(`sets Game.move  to Shells' prop move`, () => {
    const component = shallow(<Game {...props} />);
    const {move} = component.instance() as Game;
    expect(component.find(Shells).prop('move')).toEqual(move);
  });

  it(`sets Game.onMouseLeave to Shells' prop onMouseLeave`, () => {
    const component = shallow(<Game {...props} />);
    const {onMouseLeave} = component.instance() as Game;
    expect(component.find(Shells).prop('onMouseLeave')).toEqual(onMouseLeave);
  });

  it(`sets Game.onChoose to Shells' prop onChoose`, () => {
    const component = shallow(<Game {...props} />);
    const {onChoose} = component.instance() as Game;
    expect(component.find(Shells).prop('onChoose')).toEqual(onChoose);
  });

  it('sets state.chosenPosition to Shells.chosenPosition', () => {
    const component = shallow(<Game {...props} />);
    expect(component.find(Shells).prop('chosenPosition')).toEqual(
      component.state('chosenPosition')
    );
    component.setState({chosenPosition: Positions.Left});
    expect(component.find(Shells).prop('chosenPosition')).toEqual(
      Positions.Left
    );
  });

  it('sets state.isWin to Shells.isWin', () => {
    const component = shallow(<Game {...props} />);
    expect(component.find(Shells).prop('isWin')).toEqual(
      component.state('isWin')
    );
    component.setState({isWin: true});
    expect(component.find(Shells).prop('isWin')).toEqual(true);
  });

  it('has static constant SCORE_NUM', () => {
    expect(Game.SCORE_NUM).toEqual(50);
  });

  it('has static constant PRIZE', () => {
    expect(Game.PRIZE).toEqual(500);
  });
});
