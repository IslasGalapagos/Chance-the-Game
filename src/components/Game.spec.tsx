/*
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  Copyright Â© 2018 Evgeny Sysoletin. All rights reserved.
*/

import {shallow} from 'enzyme';
import * as testRenderer from 'react-test-renderer';
import * as React from 'react';
import Game, {Positions} from './Game';
import Head from './Head';
import Shells from './Shells';

const positionsArr: Positions[] = [
  Positions.Left,
  Positions.Center,
  Positions.Right
];

describe('<Game/>', () => {
  it('renders correctly', () => {
    const treeAndStyles = testRenderer.create(<Game />).toJSON();
    expect(treeAndStyles).toMatchSnapshot();
  });

  // STATE
  it(`sets 'center' to state.position by default`, () => {
    const component = shallow(<Game />);
    expect(component.state('position')).toEqual(Positions.Center);
  });

  it(`sets false to state.isMouseEnter by default`, () => {
    const component = shallow(<Game />);
    expect(component.state('isMouseEnter')).toEqual(false);
  });

  it(`sets null to state.chosenPosition by default`, () => {
    const component = shallow(<Game />);
    expect(component.state('chosenPosition')).toBeNull();
  });

  it(`sets null to state.isWin by default`, () => {
    const component = shallow(<Game />);
    expect(component.state('isWin')).toEqual(null);
  });

  // METHODS
  it('has move that changes state.position depends on arg & sets true to state.isMouseEnter', () => {
    const component = shallow(<Game />);
    const {move} = component.instance() as Game;
    move(Positions.Left);
    expect(component.state('position')).toEqual(Positions.Left);
    expect(component.state('isMouseEnter')).toEqual(true);
  });

  it('has onMouseLeave that sets false to state.isMouseEnter', () => {
    const component = shallow(<Game />);
    const {onMouseLeave} = component.instance() as Game;
    component.setState({isMouseEnter: true});
    onMouseLeave();
    expect(component.state('isMouseEnter')).toEqual(false);
  });

  it('has random that randomly returns one of Positions', () => {
    const component = shallow(<Game />);
    const {random} = component.instance() as Game;
    const randomsArr: Positions[] = Array.apply(null, {length: 5})
      .map(() => random())
      .sort();

    randomsArr.forEach((x: Positions) => {
      expect(positionsArr).toContain(x);
    });
    expect(randomsArr[0]).not.toEqual(randomsArr[4]);
  });

  it(`has onChoose that takes user's choice, determines win or fail by random & changes state for some time`, () => {
    jest.useFakeTimers();

    const spyRandom = jest.spyOn(Game.prototype, 'random');
    const component = shallow(<Game />);
    const {onChoose} = component.instance() as Game;

    expect(spyRandom).not.toBeCalled();

    onChoose(Positions.Left);
    expect(spyRandom).toBeCalledTimes(1);
    expect(component.state('chosenPosition')).toEqual(Positions.Left);
    expect(component.state('isWin')).toEqual(expect.any(Boolean));

    jest.runAllTimers();

    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 800);
    expect(component.state('chosenPosition')).toBeNull();
    expect(component.state('isWin')).toBeNull();

    jest.restoreAllMocks();
  });

  // HEAD
  it(`sets state.isWin to Head's prop isWin`, () => {
    const component = shallow(<Game />);
    expect(component.find(Head).prop('isWin')).toEqual(
      component.state('isWin')
    );
    component.setState({isWin: true});
    expect(component.find(Head).prop('isWin')).toEqual(true);
  });

  it(`sets state.position to Head's prop position`, () => {
    const component = shallow(<Game />);
    expect(component.find(Head).prop('position')).toEqual(
      component.state('position')
    );
    component.setState({position: Positions.Left});
    expect(component.find(Head).prop('position')).toEqual(Positions.Left);
  });

  it(`sets state.isMouseEnter to Head's prop isMouseEnter`, () => {
    const component = shallow(<Game />);
    expect(component.find(Head).prop('isMouseEnter')).toEqual(
      component.state('isMouseEnter')
    );
    component.setState({isMouseEnter: true});
    expect(component.find(Head).prop('isMouseEnter')).toEqual(true);
  });

  // SHELLS
  it(`sets Game.move  to Shells' prop move`, () => {
    const component = shallow(<Game />);
    const {move} = component.instance() as Game;
    expect(component.find(Shells).prop('move')).toEqual(move);
  });

  it(`sets Game.onMouseLeave to Shells' prop onMouseLeave`, () => {
    const component = shallow(<Game />);
    const {onMouseLeave} = component.instance() as Game;
    expect(component.find(Shells).prop('onMouseLeave')).toEqual(onMouseLeave);
  });

  it(`sets Game.onChoose to Shells' prop onChoose`, () => {
    const component = shallow(<Game />);
    const {onChoose} = component.instance() as Game;
    expect(component.find(Shells).prop('onChoose')).toEqual(onChoose);
  });

  it('sets state.chosenPosition to Shells.chosenPosition', () => {
    const component = shallow(<Game />);
    expect(component.find(Shells).prop('chosenPosition')).toEqual(
      component.state('chosenPosition')
    );
    component.setState({chosenPosition: Positions.Left});
    expect(component.find(Shells).prop('chosenPosition')).toEqual(
      Positions.Left
    );
  });

  it('sets state.isWin to Shells.isWin', () => {
    const component = shallow(<Game />);
    expect(component.find(Shells).prop('isWin')).toEqual(component.state('isWin'));
    component.setState({isWin: true});
    expect(component.find(Shells).prop('isWin')).toEqual(true);
  });
});
