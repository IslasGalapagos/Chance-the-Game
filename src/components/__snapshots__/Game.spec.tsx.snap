// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game/> renders correctly 1`] = `
.emotion-5 {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.emotion-0 {
  position: relative;
  left: 0;
  margin-bottom: 85px;
  -webkit-transition: left 0.2s ease;
  transition: left 0.2s ease;
  -webkit-animation: animation-pce971 1s ease infinite;
  animation: animation-pce971 1s ease infinite;
}

.emotion-4 {
  position: relative;
  display: block;
}

.emotion-1 {
  display: inline-block;
  vertical-align: bottom;
  cursor: pointer;
}

.emotion-1:hover {
  -webkit-animation: animation-qym1sx 0.2s ease infinite;
  animation: animation-qym1sx 0.2s ease infinite;
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}

.emotion-2 {
  margin: 0 50px;
  display: inline-block;
  vertical-align: bottom;
  cursor: pointer;
}

.emotion-2:hover {
  -webkit-animation: animation-qym1sx 0.2s ease infinite;
  animation: animation-qym1sx 0.2s ease infinite;
  -webkit-animation-delay: 0.2s;
  animation-delay: 0.2s;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-0"
  >
    <img
      alt="head"
      src="/images/head-default.svg"
    />
  </div>
  <div
    className="emotion-4"
  >
    <img
      alt="shell № 1"
      className="emotion-1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="./images/shell-1.svg"
    />
    <img
      alt="shell № 2"
      className="emotion-2"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="./images/shell-2.svg"
    />
    <img
      alt="shell № 3"
      className="emotion-1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="./images/shell-3.svg"
    />
  </div>
</div>
`;
